R/summariseClinicalRecords.R

```diff
@@ -71,168 +71,292 @@ summariseClinicalRecords <- function(cdm,
-  omopTable <- restrictStudyPeriod(omopTable, dateRange)
-  omopTable <- sampleOmopTable(omopTable, sample)
-  if(omopgenerics::isTableEmpty(omopTable)) {
-    return(omopgenerics::emptySummarisedResult(settings = createSettings(result_type = "summarise_clinical_records", study_period = dateRange)))
-  }
+  # get strata
+  strata <- c(
+    list(character()),
+    omopgenerics::combineStrata(strataCols(sex = sex, ageGroup = ageGroup))
+  )
 
-  omopTable <- filterPersonId(omopTable) |>
-    addStrataToOmopTable(date, ageGroup, sex)
+  # create denominator for record count
+  den <- denominator(
+    cdm = cdm,
+    sex = sex,
+    ageGroup = ageGroup,
+    name = omopgenerics::uniqueTableName(prefix)
+  )
 
-  if ("observation_period" == omopTableName) {
-    if (standardConcept) {
-      if (!missing(standardConcept)) {
-        cli::cli_inform("standardConcept turned to FALSE for observation_period OMOP table", call = call)
+  set <- createSettings(
+    result_type = "summarise_clinical_records", study_period = dateRange
+  )
+
+  result <- purrr::map(omopTableName, \(table) {
+    # check that table is not empty
+    omopTable <- dplyr::ungroup(cdm[[table]])
+    if (omopgenerics::isTableEmpty(omopTable)) {
+      cli::cli_warn("{table} is empty.")
+      return(omopgenerics::emptySummarisedResult())
+    }
+    prefix <- omopgenerics::tmpPrefix()
+
+    # warn if observation_period
+    if ("observation_period" == table) {
+      if (warnStandardConcept) {
+        "standardConcept turned to FALSE for observation_period OMOP table" |>
+          rlang::set_names("i") |>
+          cli::cli_inform()
       }
       standardConcept <- FALSE
-    }
-    if (sourceVocabulary) {
-      if (!missing(sourceVocabulary)) {
-        cli::cli_inform("sourceVocabulary turned to FALSE for observation_period OMOP table", call = call)
+      if (warnSourceVocabulary) {
+        "sourceVocabulary turned to FALSE for observation_period OMOP table" |>
+          rlang::set_names("i") |>
+          cli::cli_inform()
       }
       sourceVocabulary <- FALSE
-    }
-    if (domainId) {
-      if (!missing(domainId)) {
-        cli::cli_inform("domainId turned to FALSE for observation_period OMOP table", call = call)
+      if (warnDomainId) {
+        "domainId turned to FALSE for observation_period OMOP table" |>
+          rlang::set_names("i") |>
+          cli::cli_inform()
       }
       domainId <- FALSE
     }
-  }
-
-  strata <- getStrataList(sex, ageGroup)
-  strata <- c(list(character()), strata)
 
-  # Counts summary ----
-  cli::cli_inform(c("i" = "Summarising {.pkg {omopTableName}} counts and records per person"))
-  result <- summariseRecordsPerPerson(
-    omopTable, date, sex, ageGroup, recordsPerPerson)|>
-    omopgenerics::newSummarisedResult(settings = createSettings(""))
+    # restrict study period
+    omopTable <- restrictStudyPeriod(omopTable, dateRange)
+    if (is.null(omopTable)) return(omopgenerics::emptySummarisedResult())
```

Comment:
should we add the settings to the empty results

Reply:
https://github.com/OHDSI/OmopSketch/blob/9c78bf4c84b7710fbf9acaf8def2d330a1451d0d/R/summariseClinicalRecords.R#L273-L274

They are added later when merged with the rest of results, return only affect inside the purrr statement